openpype/hosts/traypublisher/plugins/create/create_csv_ingest.py

```diff
@@ -0,0 +1,767 @@
+        )
+
+        csv_instance = CreatedInstance(
+            self.family, subset_name, instance_data, self
+        )
+        self._store_new_instance(csv_instance)
+
+        csv_instance["csvFileData"] = {
+            "filename": filename,
+            "staging_dir": staging_dir,
+        }
+
+        # from special function get all data from csv file and convert them
+        # to new instances
+        csv_data_for_instances = self._get_data_from_csv(
+            staging_dir, filename)
+
+        # create instances from csv data via self function
+        self._create_instances_from_csv_data(
+            csv_data_for_instances, staging_dir
+        )
+
+    def _create_instances_from_csv_data(
+        self,
+        csv_data_for_instances,
+        staging_dir
+    ):
+        """Create instances from csv data"""
+
+        for asset_name, _data in csv_data_for_instances.items():
+            asset_doc = _data["asset_doc"]
+            vendor_name = _data["vendor_name"]
+            products = _data["products"]
+
+            for instance_name, product_data in products.items():
+                # get important instance variables
+                task_name = product_data["task_name"]
+                variant_name = product_data["variant_name"]
+                product_type = product_data["product_type"]
+                version = product_data["version"]
+
+                # create subset/product name
+                product_name = get_subset_name(
+                    product_type,
+                    variant_name,
+                    task_name,
+                    asset_doc,
+                )
+
+                # make sure frame start/end is inherited from csv columns
+                # if they exists or inherit them from asset doc for case where
+                # columns are missing because only mov or mp4 files needs to be
+                # published
+                frame_start = None
+                frame_end = None
+                handle_start = None
+                handle_end = None
+                comment = None
+                intent = None
+                is_calculated = False
+                for filepath, repre_data in product_data["representations"].items():  # noqa: E501
+                    if not comment and repre_data["notes"]:
+                        comment = repre_data["notes"]
+                    if not intent and repre_data["intent"]:
+                        intent = repre_data["intent"]
+                        if intent:
+                            comment = f"{intent}: {comment}"
+
+                    extension = os.path.splitext(filepath)[-1]
+                    # INFO: mp4 representation doesn't have to have frameStart
+                    # and frameEnd columns
+                    if extension in [".mp4"]:
+                        continue
+
+                    if (
+                        frame_start is None
+                        and repre_data["frameStart"] is not None
```

Comment:
Is the second condition relevant?
```python
if frame_start is None:
    frame_start = repre_data["frameStart"]
```

Less code, result same.

Reply:
The **repre_data** can contain key `frameStart` with None so this needs to be check specifically. For more information on why is `frameStart` even included if None is set, look to `openpype\hosts\traypublisher\plugins\create\create_csv_ingest.py` line 632.